### PR TITLE
Desktop: Update Title-Bar

### PR DESCRIPTION
**Pull request by ruppysuppy**
Removes the electron window frame and replaces the default title bar with a custom made title bar

## Demo

Custom Title Bar Demo

---

Related to: #2689

### CODE REVIEW

File: packages/app-desktop/ElectronAppWrapper.ts [hunk 1]
```diff
@@ -204,6 +205,23 @@ export default class ElectronAppWrapper {
...
 
+		ipcMain.on('minimize-handler', () => this.win_.minimize());
+
+		ipcMain.on('maximize-handler', () => this.win_.isMaximized() ?
+			this.win_.restore() : this.win_.maximize());
+
+		ipcMain.on('close-window-handler', () => this.win_.close());
+		ipcMain.on('display-app-menu', function(_: any, args: {x: number; y: number}) {
+			const menu = Menu.getApplicationMenu();
+			if (process.platform === 'win32') {
```
Comment (CalebJohn) [inline packages/app-desktop/ElectronAppWrapper.ts:216]: What about other platforms? Same question for TitleBar.tsx

**Comment by CalebJohn** [comment]
Can I ask why you decided to put all the menu options under a dropdown? Is this normal behaviour for electron apps?
Also could you show an example of how this looks in dark mode? A screenshot is fine?

**Comment by tessus** [comment]
On macOS Joplin now looks like a Windows application. The menu bar is way too wide vertically and what's with the red X when I hover over it? The text color is black on gray and the info `(DEV)` is missing, when I'm running the dev version.

image

I'm sorry, but I really hope this does not get merged.

image

I like thhe other one better:

image

**Comment by ruppysuppy** [comment]
I changed it to use the custom title bar only in windows & the text color as well. But I am using the window's title for the title bar text, from where can I get the actual title?

Windows Dark Mode

File: packages/app-desktop/ElectronAppWrapper.ts (hunk 1 above)
Comment (ruppysuppy) [reply packages/app-desktop/ElectronAppWrapper.ts:216]: The menu appears in Menu Bar at the top of the screen for other platforms

Comment (CalebJohn) [reply packages/app-desktop/ElectronAppWrapper.ts:216]: I guess my question is, why are you checking for win32 here?

Comment (ruppysuppy) [reply packages/app-desktop/ElectronAppWrapper.ts:216]: To check if the app is being run on a Windows system

Comment (CalebJohn) [reply packages/app-desktop/ElectronAppWrapper.ts:216]: What about Linux systems? They use the menu bar just like Windows. 

I see that you've now disabled this upgrade on non-windows systems so it's not necessary to change anymore.

**Comment by laurent22** [comment]
Thanks for the pull request but as per [the guidelines](https://joplinapp.org/gsoc2021/index/#how-to-create-your-first-pull-request) we need an associated issue and clear technical spec before accepting a new feature. We basically need to have a consensus about what needs to be implemented, and it needs to be done before creating the pull request.

**Pull request closed without merging**